spyder/widgets/explorer.py

```diff
@@ -1201,16 +1202,9 @@ def __init__(self, parent=None, name_filters=['*.py', '*.pyw'],
             widget.setIconSize(QSize(16, 16))
 
         # Layouts
-        blayout = QHBoxLayout()
-        blayout.addWidget(button_previous)
-        blayout.addWidget(button_next)
-        blayout.addWidget(button_parent)
-        blayout.addStretch()
-        blayout.addWidget(self.button_menu)
-
-        layout = QVBoxLayout()
-        layout.addLayout(blayout)
-        layout.addWidget(self.treewidget)
+        controls = [button_previous, button_next, button_parent, 'stretch',

```

Comment:
I don't like the `'stretch'` idea. I think it is better to just leave this outside of the control_layout func.